Sources/AsyncHTTPClient/ConnectionPool/State Machine/HTTPConnectionPool+RequestQueue.swift

```diff
@@ -69,6 +69,21 @@ extension HTTPConnectionPool {
                 return self.generalPurposeQueue.popFirst()
             }
         }
+        
+        /// removes up to `max` requests from the queue for the given `eventLoop` and returns them.
+        /// - Parameters:
+        ///   - max: maximum number of requests to pop
+        ///   - eventLoop: required event loop of the request
+        /// - Returns: requests for the given `eventLoop`
+        mutating func popFirst(max: Int, for eventLoop: EventLoop? = nil) -> [Request] {
+            if let eventLoop = eventLoop {
+                return self.withEventLoopQueue(for: eventLoop.id) { queue in
+                    queue.popFirst(max: max)
+                }
```

Comment:
if max isn't reached here, we should further pop requests from general purpose queue

Reply:
already handled in `HTTP2StateMaschine`